### PR TITLE
BigQuery Stream doc update

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## What do these changes do?

> Updated user permissions
> Restructured the doc for better information flow.
> Reduced unnecessary callouts and improved language in some sections.

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

### CODE REVIEW

File: docs/destinations/streaming-destinations/bigquery-stream.mdx [hunk 1]
```diff
@@ -1,91 +1,75 @@
...
 ---
 
-[Google BigQuery](https://cloud.google.com/bigquery) lets you stream your event data by leveraging its [streaming API](https://cloud.google.com/bigquery/docs/samples/bigquery-table-insert-rows). With this approach, you can insert data into BigQuery in near real-time and have it ready for analysis in no time.
+[Google BigQuery](https://cloud.google.com/bigquery) lets you stream your event data using its [streaming API](https://cloud.google.com/bigquery/docs/samples/bigquery-table-insert-rows). With this API, you can insert data into BigQuery in near real-time and have it ready for analysis in no time.
 
-RudderStack supports BigQuery Stream as a destination to which you can steam your event data seamlessly.
+Find the open source transformer code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/src/v0/destinations/bqstream">GitHub repository</a>.
 
-## Getting started
+## Creating a service account
```
Comment (akashrpo) [inline docs/destinations/streaming-destinations/bigquery-stream.mdx:10]: Are we not placing the details of the integration at the top any longer (`#Getting started` section where it displays all the modes)

Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/bigquery-stream.mdx:10]: @akashrpo it's there below the service account section. I thought of moving the section at the top as it's a prerequisite, before the user starts configuring the destination.

Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/bigquery-stream.mdx:10]: You can check the [preview](https://rudderstack-docs-git-bqstream-doc-updates-rudder-stack.vercel.app/destinations/streaming-destinations/bigquery-stream/#getting-started)

Comment (akashrpo) [reply docs/destinations/streaming-destinations/bigquery-stream.mdx:10]: IMO the landing page seems to be more aligned towards configuration than getting to know about the integration in short

**Review by ItsSudip — APPROVED** [review]

File: docs/destinations/streaming-destinations/bigquery-stream.mdx (hunk 1 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/bigquery-stream.mdx:10]: @akashrpo Do you think the following structure is more helpful, then?

- Getting Started (with a note highlighting that the user needs to set up a service account first, with a link to the section)
- Connection settings
- Sending events
- Deduplicating data
- Dynamically configuring `insertId`
- Creating service account
- Troubleshooting

File: docs/destinations/streaming-destinations/bigquery-stream.mdx [hunk 2]
```diff
@@ -1,91 +1,75 @@
...
-Consider the following table schema:
-
-<img src="../../assets/event-stream-destinations/bqstream-5.png" alt="BigQuery table schema" />
-
-When sending an`Insert Product` event to BigQuery, the `productId` can be used to uniquely identify the product. Hence it can be passed as the `insertId`. BigQuery performs the de-duplication of the data based on this `productId`. 
-
 ## Sending events to BigQuery Stream
 
-You can send only `track` events to BigQuery Stream via RudderStack. Also, RudderStack currently does not support batching these events.
+RudderStack supports sending only <Link to="/event-spec/standard-events/track">`track`</Link> events to BigQuery Stream. Also, it currently **does not support** batching these events.
```
Comment (akashrpo) [inline docs/destinations/streaming-destinations/bigquery-stream.mdx:69]:
```suggestion
RudderStack supports sending only <Link to="/event-spec/standard-events/track">`track`</Link> events to BigQuery Stream.
```
We can remove this statement as we support batching now

File: docs/destinations/streaming-destinations/bigquery-stream.mdx (hunk 1 above)
Comment (akashrpo) [reply docs/destinations/streaming-destinations/bigquery-stream.mdx:10]: Looks good

File: docs/destinations/streaming-destinations/bigquery-stream.mdx (hunk 2 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/bigquery-stream.mdx:69]: Done.

File: docs/destinations/streaming-destinations/bigquery-stream.mdx (hunk 1 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/bigquery-stream.mdx]: Thanks - I've incorporated this structure.

**Review by akashrpo — APPROVED** [review]

File: docs/destinations/streaming-destinations/bigquery-stream.mdx [hunk 3]
```diff
@@ -1,91 +1,53 @@
...
-Consider the following table schema:
-
-<img src="../../assets/event-stream-destinations/bqstream-5.png" alt="BigQuery table schema" />
-
-When sending an`Insert Product` event to BigQuery, the `productId` can be used to uniquely identify the product. Hence it can be passed as the `insertId`. BigQuery performs the de-duplication of the data based on this `productId`. 
-
 ## Sending events to BigQuery Stream
 
-You can send only `track` events to BigQuery Stream via RudderStack. Also, RudderStack currently does not support batching these events.
+RudderStack supports sending only <Link to="/event-spec/standard-events/track">`track`</Link> events to BigQuery Stream.
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/bigquery-stream.mdx:47]:
```suggestion
RudderStack supports sending only <Link to="/event-spec/standard-events/track">`track`</Link> events to BigQuery Stream. The `track` event payload format should match the table schema specified in the RudderStack dashboard.
```

> table schema specified in the RudderStack dashboard..

not sure where is this in the dashboard

File: docs/destinations/streaming-destinations/bigquery-stream.mdx [hunk 4]
```diff
@@ -1,91 +1,53 @@
...
 
-<div class="warningBlock">
-Make sure the <code class="inline-code">track</code> event payload format matches the table schema specified in the RudderStack dashboard.
-</div>
-
-<div class="infoBlock">
-
-RudderStack currently does not support the <code class="inline-code">templateSupportSuffix</code> feature which creates a table schema during a streaming insert action.
-</div>
+- Make sure the `track` event payload format matches the table schema specified in the RudderStack dashboard.
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/bigquery-stream.mdx:49]:
```suggestion
```

File: docs/destinations/streaming-destinations/bigquery-stream.mdx [hunk 5]
```diff
@@ -166,7 +116,7 @@ Consider the following payload:
 }
 ```
 
-In the above case, de-duplication **will not be applicable** as the event properties do not contain `<some_value>` present in the payload.
+In the above case, deduplication **will not be applicable** as the event properties do not contain `<some_value>` present in the payload.
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/bigquery-stream.mdx:119]:
```suggestion
In the above case, deduplication **is not applicable** as the event properties do not contain `<some_value>` present in the payload.
```

File: docs/destinations/streaming-destinations/bigquery-stream.mdx [hunk 6]
```diff
@@ -1,91 +1,53 @@
...
-
-RudderStack currently supports only <strong>numeric</strong> or <strong>string</strong> values as `insertId`.
-</div>
+- **Insert ID**: This is an **optional** field. Enter the insert ID used by Google to deduplicate the data sent to BigQuery. For more information on this setting, refer to the <Link to="#deduplicating-data">Deduplicating data</Link> section.
+- **Credentials**: Enter the contents of the credentials JSON you downloaded after <Link to="#creating-a-service-account">creating your service account</Link>.
 
 <div class="infoBlock">
-
-For more information on the de-duplication process in BigQuery, refer to this <a href="https://cloud.google.com/bigquery/streaming-data-into-bigquery#dataconsistency">BigQuery documentation</a>.
+Create a service account in your GCP Project for RudderStack with the <strong>BigQuery Data Editor</strong> role, which has the required access to write to BigQuery.
```
Comment (ishachopra0401) [inline docs/destinations/streaming-destinations/bigquery-stream.mdx:42]: Bit confused if it is the same as "Creating a service account" section below or different

Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/bigquery-stream.mdx:42]: It's the same - good catch, will cross-reference the section here as well.

File: docs/destinations/streaming-destinations/bigquery-stream.mdx [hunk 7]
```diff
@@ -1,91 +1,53 @@
...
-## Insert ID
-
-Google leverages the `insertId` to de-duplicate the data sent to BigQuery. `insertId` is essentially an event property that uniquely identifies an event.
-
-<div class="warningBlock">
-
-RudderStack currently supports only <strong>numeric</strong> or <strong>string</strong> values as `insertId`.
-</div>
+- **Insert ID**: This is an **optional** field. Enter the insert ID used by Google to deduplicate the data sent to BigQuery. For more information on this setting, refer to the <Link to="#deduplicating-data">Deduplicating data</Link> section.
+- **Credentials**: Enter the contents of the credentials JSON you downloaded after <Link to="#creating-a-service-account">creating your service account</Link>.
```
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/bigquery-stream.mdx:39]: Actually, there's a lot more information in the JSON file than just the private key; all the contents must be pasted in this field as is. So the original sentence should be retained.

File: docs/destinations/streaming-destinations/bigquery-stream.mdx (hunk 3 above)
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/bigquery-stream.mdx:47]: we specify the ID of the table - I'll tweak the sentence to have no scope for confusion.

File: docs/destinations/streaming-destinations/bigquery-stream.mdx [hunk 8]
```diff
@@ -1,91 +1,53 @@
...
-<div class="warningBlock">
-Make sure the <code class="inline-code">track</code> event payload format matches the table schema specified in the RudderStack dashboard.
-</div>
-
-<div class="infoBlock">
-
-RudderStack currently does not support the <code class="inline-code">templateSupportSuffix</code> feature which creates a table schema during a streaming insert action.
-</div>
+- Make sure the `track` event payload format matches the table schema specified in the RudderStack dashboard.
+- RudderStack currently does not support the `templateSupportSuffix` feature which creates a table schema during a streaming insert action.
```
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply docs/destinations/streaming-destinations/bigquery-stream.mdx:50]: Yes - it was originally a callout anyway - I tweaked it more along the lines of "Also note that..." followed by the bullets. But your suggestion makes sense.

**Review by ishachopra0401 — APPROVED** [review]

File: docs/destinations/streaming-destinations/bigquery-stream.mdx [hunk 9]
```diff
@@ -1,89 +1,48 @@
...
 | Connection Mode | Web       | Mobile    | Server   |
 | :------------------ | :------------ | :------------ | :------------ |
 | **Device mode**     | -             | -             | -             |
 | **Cloud mode**      | **Supported** | **Supported** | **Supported** |
 
-<div class="infoBlock">
-To learn more about the difference between cloud mode and device mode in RudderStack, refer to the <Link to="/destinations/rudderstack-connection-modes/">RudderStack Connection Modes</Link> guide.
-</div>
-
 Once you've confirmed that the source platform supports sending events to BigQuery Stream, follow these steps:
```
Comment (ameypv-rudder) [inline docs/destinations/streaming-destinations/bigquery-stream.mdx:23]:
```suggestion
Once you have confirmed that the source platform supports sending events to BigQuery Stream, follow these steps:
```

**Pull request merged**